backend/src/main/scala/cromwell/backend/wdl/ReadLikeFunctions.scala

```diff
@@ -6,10 +6,28 @@ import wdl4s.expression.WdlStandardLibraryFunctions
 import wdl4s.parser.MemoryUnit
 import wdl4s.types.{WdlArrayType, WdlFileType, WdlObjectType, WdlStringType}
 import wdl4s.values._
+import cats._, implicits._
 
 import scala.util.{Failure, Success, Try}
 
-trait ReadLikeFunctions extends PathFactory { this: WdlStandardLibraryFunctions =>
+trait FileSizeLimitationConfig {
+  val readLinesLimit = 128000
+  val readBoolLimit = 5
+  val readIntLimit = 19
+  val readFloatLimit = 50
+  val readStringLimit = 128000
+  val readJsonLimit = 128000
+  val readTsvLimit = 128000
+  val readMapLimit = 128000
+  val readObjectLimit = 128000
+}
+
+case class FileSizeTooBig(override val getMessage: String) extends Exception
+
+trait ReadLikeFunctions extends PathFactory with FileSizeLimitationConfig { this: WdlStandardLibraryFunctions =>
+
+  def fileSize: WdlValue=> Try[Long] = 
+    w => Try(buildPath(w.valueString).size)
```

Comment:
Yeah, IIRC my comment came  before the one that @kcibul made. I was trying to avoid the situation where a giant file is pulled across the network and then we just read a few bytes, so in that case I agree w/ what @Horneth says. However Kristian's comment is also right, I agree that it should be an error, so size would need to be checked anyways.